src/Workspaces/Core/Portable/PatternMatching/PatternMatch.cs

```diff
@@ -61,16 +61,23 @@ public int CompareTo(PatternMatch other)
             => CompareTo(other, ignoreCase: false);
 
         public int CompareTo(PatternMatch other, bool ignoreCase)
-            => ComparerWithState.CompareTo(this, other, ignoreCase, s_comparers);
+        {
+            // Compare types
+            var comparison = this.Kind - other.Kind;
+            if (comparison != 0)
+                return comparison;
+
+            // Compare cases
+            if (!ignoreCase)
+            {
+                comparison = (!this.IsCaseSensitive).CompareTo(!other.IsCaseSensitive);
+                if (comparison != 0)
+                    return comparison;
+            }
 
-        private static readonly ImmutableArray<Func<PatternMatch, bool, IComparable>> s_comparers =
-            ImmutableArray.Create<Func<PatternMatch, bool, IComparable>>(
-                // Compare types
-                (p, b) => p.Kind,
```

Comment:
💡 Full parameter names here would help clarity.

💡 Explicit `static` here would help clarity.

Reply:
this code is removed